pkg/kudoctl/util/kudo/kudo.go

```diff
@@ -292,6 +292,26 @@ func (c *Client) WaitForInstance(name, namespace string, oldInstance *kudoapi.In
 	})
 }
 
+// WaitForInstanceDeleted waits for instance to be removed from the cluster.
+func (c *Client) WaitForInstanceDeleted(name, namespace string, timeout time.Duration) error {
+	interval := 1 * time.Second
+
+	return wait.PollImmediate(interval, timeout, func() (done bool, err error) {
+		instance, err := c.GetInstance(name, namespace)
+		if err != nil {
```

Comment:
Ah I hate that we redefined the error not found to not be returned 🙈 I originally wanted to reject because it's not handled.

Reply:
My first iteration of the code handled that case here, resulting in the unit test to fail with a timeout because there won't be a not-found-error 😆 🙈

Reply:
Might it be worth to unify that with the `health.isDeleted`? That code currently works on a `runtime.Object`, not just a namespace/name, but maybe it's worth to generalise? 

I'm ok with this as well though, but I also stumbled over the missing `NotFound` handling ;)

Reply:
It would make sense to unify API usage in KUDO. Controller code is `runtime.Object` based while the CLI uses the generated API.